homeassistant/components/backup/config.py

```diff
@@ -62,16 +122,150 @@ async def load(self) -> None:
         if stored:
             self.data = BackupConfigData.from_dict(stored)
 
-    async def save(self) -> None:
+        self.data.schedule.apply(self._manager)
+
+    @callback
+    def save(self) -> None:
         """Save config."""
-        await self._store.async_save(self.data.to_dict())
+        self._store.async_delay_save(self._data_to_save, DELAY_SAVE)
+
+    @callback
+    def _data_to_save(self) -> StoredBackupConfig:
+        """Return data to save."""
+        return self.data.to_dict()
 
     async def update(
-        self, *, max_copies: int | None | UndefinedType = UNDEFINED
+        self,
+        *,
+        addons_included: list[str] | None | UndefinedType = UNDEFINED,
+        agent_ids: list[str] | UndefinedType = UNDEFINED,
+        database_included: bool | UndefinedType = UNDEFINED,
+        folders_included: list[str] | None | UndefinedType = UNDEFINED,
+        max_copies: int | None | UndefinedType = UNDEFINED,
+        name: str | None | UndefinedType = UNDEFINED,
+        password: str | None | UndefinedType = UNDEFINED,
+        schedule: ScheduleParameterDict | UndefinedType = UNDEFINED,
```

Comment:
Can we try making this a [typed dic](https://typing.readthedocs.io/en/latest/spec/callables.html#unpack-for-keyword-arguments)t to get rid of some of the repetition?